src/math/power.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+
+/* Function Declarations */
+static int32_t div_s32s64(int64_t denominator);
+
+/* Function Definitions */
+/**
+ * Arguments	: int64_t denominator
+ *
+ * Return Type	: int32_t
+ */
+static inline int32_t div_s32s64(int64_t denominator)
+{
+	uint64_t b_denominator;
+	int32_t quotient;
+
+	if (denominator == 0) {
+		quotient = INT32_MAX;
+	} else {
+		if (denominator < 0)
+			b_denominator = ~(uint64_t)denominator + 1;
+		else
+			b_denominator = (uint64_t)denominator;
+
+		b_denominator = 1125899906842624ULL / b_denominator;
```

Comment:
Magic number

Reply:
1125899906842624 is 0x4000000000000. XCC compiler throws warning "integer constant is too large for 'unsigned long' . So **1125899906842624ULL** magic

Reply:
Doesn't LL work with hex too?

Reply:
It could still be a #define with descriptive name, is it 2^50 ?

Reply:
Yup, It works :-)